src/sentry/static/sentry/app/views/projectFilters.jsx

```diff
@@ -390,11 +396,94 @@ const ProjectFilters = React.createClass({
   },
 
   componentDidUpdate(prevProps) {
-    if (!this.state.loading && !this.state.stats) {
-      this.processStatsData();
+    if (!this.state.loading && !this.state.formattedData) {
+      this.render();
     }
   },
 
+  getStatOpts() {
+    return(
+      this.getProjectFeatures().has('additional-data-filters') ?
+      {
+        'ip-address': 'IP Address',
+        'release-version': 'Release',
+        'error-message': 'Error Message',
+        'browser-extensions': 'Browser Extension',
+        'legacy-browsers': 'Legacy Browser',
+        localhost: 'Localhost',
+        'web-crawlers': 'Web Crawler',
+        'invalid-csp': 'Invalid CSP',
+        cors: 'CORS'
+      } :
+      {'blacklisted':"Filtered Events"}
+    );
+  },
+
+  formatData(rawData) {
+    return Object.keys(this.getStatOpts()).map(stat => {
+      return {
+        data: rawData[stat].map(([x, y]) => {
+          if (y > 0) {
+            this.setState({blankStats: false});
+          }
+          return {x, y};
+        }),
+        label: this.getStatOpts()[stat],
+        statName: stat
+      };
+    });
+  },
+
+  getFilterStats() {
+    let stat_options = Object.keys(this.getStatOpts());
+    let {orgId, projectId} = this.props.params;
+    let statEndpoint = `/projects/${orgId}/${projectId}/stats/`;
+    let query = {
+      since: this.state.querySince,
+      until: this.state.queryUntil,
+      resolution: '1d'
+    };
+    $.when
+      .apply(
+        $,
+        stat_options.map(stat => {
+          let deferred = $.Deferred();
+          this.api.request(statEndpoint, {
+            query: Object.assign({stat: stat}, query),
+            success: deferred.resolve.bind(deferred),
+            error: deferred.reject.bind(deferred)
+          });
+          return deferred;
+        })
+      )
+      .done(
+        function() {
```

Comment:
When passing arbitrary length-array values as `arguments`, I try to document them as function params.

e.g.

```javascript
  function(/* statOption1, statOption2, ... statOptionN */)
```

Something like that.